### PR TITLE
fix(SelectiveMerge):always override with df2's column (if exists)

### PR DESCRIPTION
**Pull request by Doe-Ed**

### CODE REVIEW

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala [hunk 1]
```diff
@@ -44,6 +45,10 @@ object SelectiveMerge {
     }
   }
 
+  def dummy(df1: DataFrame): Unit = {
```
Comment (doronporat) [inline src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:48]: Why dummy?

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala [hunk 2]
```diff
@@ -88,31 +82,24 @@ object SelectiveMerge {
...
           val colName = currColumn.expr.asInstanceOf[NamedExpression].name
-          // Column is a part of df1 and doesn't belong to the join keys.
-          if (isMemberOfDf1(df1.schema, colName) && !joinKeys.contains(colName)) {
-            val origColName = colName.substring(colRenamePrefixLen)
-            when(mergedDf(colName).isNotNull, mergedDf(colName).cast(df1.schema(origColName).dataType))
-              .otherwise(df1(origColName))
-              .alias(origColName)
+          // Belongs to DF2, override.
+          if (colName.startsWith(colRenamePrefix)) {
+            mergedDf(colName).alias(colName.substring(colRenamePrefixLen))
```
Comment (doronporat) [inline src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:94]: btw, you could use split function to cut off everything before the first _ including, and then you don't need to keep the prefix length

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala [hunk 3]
```diff
@@ -88,31 +82,24 @@ object SelectiveMerge {
...
+          }
+          // Is the join key(s)
+          else if (joinKeys.contains(colName)) {
+            mergedDf(colName)
           }
-          // Column doesn't belong to df1 or is join key
+          // Only exists in DF1.
           else {
```
Comment (doronporat) [inline src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:101]: What do you think about logging at the end all the column names you used from df2 to override df1 with, and the number of records you didn't match in df2 at all? if you have the number of records from df2 you didn't find in df1 it's also interesting.

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala (hunk 1 above)
Comment (Doe-Ed) [reply src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:48]: So I can recognize my docker build, removing :)

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala (hunk 2 above)
Comment (Doe-Ed) [reply src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:94]: done

File: src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala (hunk 3 above)
Comment (Doe-Ed) [reply src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala:101]: Done. example of the print for the first example:
20/05/18 11:20:36 INFO SelectiveMerge$: DF1 columns which will be overriden: List(salary, age)
20/05/18 11:20:36 INFO SelectiveMerge$: DF1 columns which are not found in DF2: List(fake, fake2)
20/05/18 11:20:36 INFO SelectiveMerge$: DF2 columns which are not found in DF1: List(bonus)

**Pull request merged**